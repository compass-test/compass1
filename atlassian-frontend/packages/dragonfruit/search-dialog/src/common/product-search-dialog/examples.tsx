import React from 'react';
import { IntlProvider } from 'react-intl';
import { ProductSearchDialog } from './product-search-dialog';
import { SharedClient } from '../clients';
import { DEFAULT_AB_TEST } from '../ab-test-provider';
import { Products } from '../product-context';

export default {
  title: 'Common Components/Product Search Dialog',
  decorators: [
    (story: () => React.ElementType) => (
      <IntlProvider locale="en">{story()}</IntlProvider>
    ),
  ],
};

const mockSharedClient: SharedClient = {
  getAbTestData: async () => DEFAULT_AB_TEST,
  getProductPermissions: async (products) => Promise.resolve(products),
};

const slowSharedClient: SharedClient = {
  getAbTestData: () =>
    new Promise((r) => setTimeout(() => r(DEFAULT_AB_TEST), 3000)),
  getProductPermissions: async (products) => Promise.resolve(products),
};

const defaultProps = {
  onClose: () => {},
  onOpen: () => {},
  isExpanded: false,
  products: [Products.confluence],
  children: (setRef: (ref: HTMLInputElement | null) => void) => (
    <input placeholder="Fully loaded component" />
  ),
  selectedTabIndex: 0,
  doProductPermissionsCheck: true,
};

export const basic = () => (
  <ProductSearchDialog {...defaultProps} sharedClient={mockSharedClient} />
);

export const slowExperiment = () => (
  <ProductSearchDialog {...defaultProps} sharedClient={slowSharedClient} />
);
