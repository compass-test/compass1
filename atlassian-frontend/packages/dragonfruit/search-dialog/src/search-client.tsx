// This is a thin wrapper around `base-search-client` to expose it externally.
// At the moment this is considered a pseudo API so be wary if making changes here to method signatures.
import * as ConfTypes from './confluence/clients/response-types';

import {
  AggregatorClient,
  SearchResponseWithTiming,
} from './common/clients/base-search-client';

export interface Config {
  /**
   * The cloud id of the instance you're searching for
   */
  cloudId: string;
  /**
   * The URL to the search backend (xpsearch-aggregator).
   * This is entirely optional and is used to override the default URL when you need to make a request to a specific xpsearch-aggregator.
   */
  aggregatorUrl?: string;
}

type AllSupportedResponseAndScopes = ConfTypes.SupportedScopedResponses;

export type PageBlogAttachmentResponse = ConfTypes.ConfItemResponse;
export type PageBlogResponse = ConfTypes.ConfItemResponse;
export type SpaceResponse = ConfTypes.ConfSpaceResponse;
export type ConfluenceUrsResponse = ConfTypes.ConfUrsPeopleResponse;
export type ConfluencePeopleResponse = ConfTypes.ConfPeopleResponse;

export type Scope = AllSupportedResponseAndScopes['id'];
export const Scope = { ...ConfTypes.Scope };

export interface SeachAnalyticParams {
  /**
   * Used to tag analytics so it can be differentiated with other consumers of search,
   * Example: 'media-editor' or 'portfolio-search'
   */
  identifier: string;
  /**
   * This is used to group together multiple searches.
   * If used this should be a UUID.
   *
   * Note that the search client does not maintain any logic as to when this is invalidated it only guarantees that this is passed
   * to the backend.
   */
  sessionId?: string;
}

interface SearchParams<SearchScope extends Scope> {
  /**
   * The query to search for.
   */
  query: string;
  /**
   * The types of entities to search for.
   */
  scopes: SearchScope[];
  /**
   * Meta information for analytic purposes.
   */
  analytics: SeachAnalyticParams;
  /**
   * The number of results to find, this applies to every scope that is being searched for.
   */
  resultLimit?: number;
}

export interface SearchClient {
  search<SearchScope extends Scope>(
    params: SearchParams<SearchScope>,
  ): Promise<
    SearchResponseWithTiming<Scope, AllSupportedResponseAndScopes, SearchScope>
  >;
}

/**
 * Creates a search client.
 *
 * This operation is considered expensive and it's recommended that the resulting client is reused.
 */
export const createClient: (config: Config) => SearchClient = ({
  cloudId,
  aggregatorUrl,
}) => {
  const aggregatorClient: AggregatorClient<
    AllSupportedResponseAndScopes,
    []
  > = new AggregatorClient({
    cloudId,
    url: aggregatorUrl || '/gateway/api/xpsearch-aggregator',
    siteMasterList: [],
  });

  async function search<SearchScope extends Scope>({
    query,
    scopes,
    resultLimit,
    analytics,
  }: SearchParams<SearchScope>) {
    return aggregatorClient.search<SearchScope>({
      query,
      context: {
        sessionId: analytics.sessionId ? analytics.sessionId : 'not-applicable',
        referrerId: null,
      },
      scopes,
      modelParams: [],
      resultLimit,
      filters: [],
      experience: `external-${analytics.identifier}`,
    });
  }

  return {
    search,
  };
};
