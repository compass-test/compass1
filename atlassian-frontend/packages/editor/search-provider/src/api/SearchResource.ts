import {
  createClient,
  SearchClient,
  Scope,
  PageBlogResponse,
  IssueResponse,
} from '@atlassian/product-search-dialog/search-client';
import {
  SearchProvider,
  QuickSearchResult,
  LinkContentType,
} from '@atlaskit/editor-common/provider-factory';
import { ResourceIdentifier, AriConstructorType } from '@atlassian/cs-ari';

const ANALYTICS_IDENTIFIER = 'editor-quick-search';

export default class SearchResource implements SearchProvider {
  private client: SearchClient;
  private scope: Scope;
  private cloudId: string;

  constructor(cloudId: string, scope: Scope, aggregatorUrl?: string) {
    switch (scope) {
      case Scope.ConfluencePageBlog:
      case Scope.JiraIssue:
        break;
      default:
        throw new Error(`Unsupported scope - ${scope}`);
    }
    this.scope = scope;
    this.cloudId = cloudId;

    const config = {
      cloudId,
      aggregatorUrl,
    };
    this.client = createClient(config);
  }

  private getAriString(opt: AriConstructorType): string {
    try {
      return new ResourceIdentifier(opt).toString();
    } catch (e) {
      return '';
    }
  }

  private confluencePageBlogResponseTransformer(
    response: PageBlogResponse | null,
  ): QuickSearchResult[] {
    if (!response) {
      return [];
    }

    return response.results.map((item) => ({
      title: item.title,
      // The `item.url` is only the relative path so we need to prefix the `item.baseUrl`
      url: item.baseUrl + item.url,
      container: item.container.title,
      contentType: this.mapToContentType(item.content && item.content.type),
      updatedTimestamp: item.lastModified,
      objectId: this.getAriString({
        resourceOwner: 'confluence',
        cloudId: this.cloudId,
        resourceType: 'content',
        resourceId: item.content && item.content.id,
      }),
    }));
  }

  private jiraIssueResponseTransformer(
    response: IssueResponse | null,
  ): QuickSearchResult[] {
    if (!response) {
      return [];
    }

    return response.results.map((item) => ({
      title: item.name,
      // The `item.url` here for Jira Issue is the absolute url
      url: item.url,
      container:
        (item.attributes.container && item.attributes.container.title) || '',
      contentType: this.mapToContentType(
        item.attributes['@type'],
        item.attributes.issueTypeName,
      ),
      updatedTimestamp: item.attributes.updated,
      objectId: this.getAriString({
        resourceOwner: 'jira',
        cloudId: this.cloudId,
        resourceType: 'issue',
        resourceId: item.id,
      }),
    }));
  }

  private mapToContentType(
    type?: 'page' | 'blogpost' | 'issue' | string,
    subType?: string,
  ): LinkContentType {
    switch (type) {
      case 'issue':
        switch (subType) {
          case 'Bug':
            return 'jira.issue.bug';
          case 'Story':
            return 'jira.issue.story';
          case 'Task':
            return 'jira.issue.task';
          default:
            return 'jira.issue';
        }
      case 'page':
        return 'confluence.page';
      case 'blogpost':
        return 'confluence.blogpost';
      default:
        return 'default';
    }
  }

  async quickSearch(
    query: string,
    limit: number,
  ): Promise<QuickSearchResult[]> {
    const result = await this.client.search({
      query,
      scopes: [this.scope],
      analytics: {
        identifier: ANALYTICS_IDENTIFIER,
      },
      resultLimit: limit,
    });

    switch (this.scope) {
      case Scope.ConfluencePageBlog: {
        const scopeResult = result.response.retrieveScope(this.scope);
        return this.confluencePageBlogResponseTransformer(scopeResult);
      }
      case Scope.JiraIssue: {
        const scopeResult = result.response.retrieveScope(this.scope);
        return this.jiraIssueResponseTransformer(scopeResult);
      }
      default:
        throw new Error(`Unsupported scope - ${this.scope}`);
    }
  }
}
