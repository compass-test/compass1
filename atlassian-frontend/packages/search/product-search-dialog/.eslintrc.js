module.exports = {
  rules: {
    'import/no-extraneous-dependencies': [
      'error',
      {
        devDependencies: [
          '**/examples.tsx',
          // __tests__ dirs inside src
          '**/__tests__/**/*.{js,ts,tsx}',
          // Storybook
          '.storybook/*',
          // Examples
          '**/examples/**/*.tsx',
          //Docs
          '**/docs/*',
          // Integration tests
          '**/analytics-integration-tests/**/*',
        ],
      },
    ],
    'import/no-internal-modules': ['warn'],
  },
};
