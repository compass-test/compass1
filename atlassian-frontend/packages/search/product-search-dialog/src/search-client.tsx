// This is a thin wrapper around `base-search-client` to expose it externally.
// At the moment this is considered a pseudo API so be wary if making changes here to method signatures.
import * as ConfTypes from './confluence/clients/response-types';
import * as JiraTypes from './jira/clients/response-types';

import {
  AggregatorClient,
  SearchResponseWithTiming,
} from './common/clients/base-search-client';

export interface Config {
  /**
   * The cloud id of the instance you're searching for
   */
  cloudId: string;
  /**
   * The URL to the search backend (xpsearch-aggregator).
   * This is entirely optional and is used to override the default URL when you need to make a request to a specific xpsearch-aggregator.
   */
  aggregatorUrl?: string;
}

type AllSupportedResponseAndScopes =
  | ConfTypes.SupportedScopedResponses
  | JiraTypes.SupportedResponseTypes;

export type PageBlogAttachmentResponse = ConfTypes.ConfItemResponse;
export type PageBlogResponse = ConfTypes.ConfItemResponse;
export type SpaceResponse = ConfTypes.ConfSpaceResponse;
export type ConfluenceUrsResponse = ConfTypes.ConfUrsPeopleResponse;
export type ConfluencePeopleResponse = ConfTypes.ConfPeopleResponse;

export type IssueResponse = JiraTypes.ServerResponse<JiraTypes.Scope.JiraIssue>;
export type ProjectResponse = JiraTypes.ServerResponse<
  JiraTypes.Scope.JiraProject
>;
export type BoardProjectFilterResponse = JiraTypes.ServerResponse<
  JiraTypes.Scope.JiraBoardProjectFilter
>;
export type JiraPeopleResponse = JiraTypes.ServerResponse<
  JiraTypes.Scope.People
>;

export type Scope = AllSupportedResponseAndScopes['id'];
export const Scope = { ...ConfTypes.Scope, ...JiraTypes.Scope };

export interface SeachAnalyticParams {
  /**
   * Used to tag analytics so it can be differentiated with other consumers of search,
   * Example: 'media-editor' or 'portfolio-search'
   */
  identifier: string;
  /**
   * This is used to group together multiple searches.
   * If used this should be a UUID.
   *
   * Note that the search client does not maintain any logic as to when this is invalidated it only guarantees that this is passed
   * to the backend.
   */
  sessionId?: string;
}

interface SearchParams<SearchScopes extends Scope[]> {
  /**
   * The query to search for.
   */
  query: string;
  /**
   * The types of entities to search for.
   */
  scopes: SearchScopes;
  /**
   * Meta information for analytic purposes.
   */
  analytics: SeachAnalyticParams;
  /**
   * The number of results to find, this applies to every scope that is being searched for.
   */
  resultLimit?: number;
}

export interface SearchClient {
  search<SearchScopes extends Scope[]>(
    params: SearchParams<SearchScopes>,
  ): Promise<
    SearchResponseWithTiming<Scope, AllSupportedResponseAndScopes, SearchScopes>
  >;
}

/**
 * Creates a search client.
 *
 * This operation is considered expensive and it's recommended that the resulting client is reused.
 */
export const createClient: (config: Config) => SearchClient = ({
  cloudId,
  aggregatorUrl,
}) => {
  const aggregatorClient: AggregatorClient<
    AllSupportedResponseAndScopes,
    []
  > = new AggregatorClient({
    cloudId,
    url: aggregatorUrl || '/gateway/api/xpsearch-aggregator',
    siteMasterList: [],
  });

  async function search<SearchScopes extends Scope[]>({
    query,
    scopes,
    resultLimit,
    analytics,
  }: SearchParams<SearchScopes>) {
    return aggregatorClient.search<SearchScopes>({
      query,
      context: {
        sessionId: analytics.sessionId ? analytics.sessionId : 'not-applicable',
        referrerId: null,
      },
      scopes,
      modelParams: [],
      resultLimit,
      filters: [],
      experience: `external-${analytics.identifier}`,
    });
  }

  return {
    search,
  };
};
