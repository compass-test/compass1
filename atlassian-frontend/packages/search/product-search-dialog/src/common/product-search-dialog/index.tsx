import * as ProductSearchDialogImports from './product-search-dialog';

export { ProductSearchDialog } from './product-search-dialog';
export type Props = ProductSearchDialogImports.Props;
