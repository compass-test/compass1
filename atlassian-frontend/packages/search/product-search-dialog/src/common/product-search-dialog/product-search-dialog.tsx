import React, { useState, useEffect } from 'react';
import { SearchCSS } from '@atlaskit/atlassian-navigation';
import {
  SearchAnchor,
  ThemeProvider,
  KeyboardHighlightProvider,
} from '@atlassian/search-dialog';
import MessagesIntlProvider from '../../common/message-intl-provider/messages-intl-provider';
import { ExperimentExposureHandler } from '../experiment-exposure-handler';
import { SearchSessionProvider } from '../../common/search-session-provider';
import { BuildVersionAnalyticContext } from '../../common/build-version-analytic-context';
import { Products, ProductProvider } from '../../common/product-context';
import { fetchPermissions } from './utils/fetch-permissions';
import { ABTestProvider } from '../ab-test-provider';
import { SharedClient } from '../clients';
import { ProductSearchInputSkeleton } from '../product-search-input-skeleton';

export interface Props {
  onClose: () => void;
  isExpanded: boolean;
  sharedClient: SharedClient;
  theme?: SearchCSS;
  children?: (
    setRef: (ref: HTMLInputElement | null) => void,
  ) => React.ReactElement | null;
  products: Products[];
  selectedTabIndex: number;
  primaryProduct?: string;
  doProductPermissionsCheck: boolean;
}

interface State {
  ref: HTMLElement | null;
}

export class ProductSearchDialogBase extends React.Component<Props, State> {
  state = {
    ref: null as HTMLElement | null,
  };

  /**
   * Hold a timeout id for the onBlur and onFocus handlers
   */
  timeout: number | undefined = undefined;

  handleKeyDown = (event: React.KeyboardEvent<HTMLDivElement>) => {
    if (
      event.key === 'Escape' &&
      !event.defaultPrevented &&
      this.props.isExpanded
    ) {
      event.preventDefault();
      this.close();
    }
  };

  onBlurHandler = () => {
    this.timeout = setTimeout(() => {
      this.close();
    });
  };

  onFocusHandler = () => {
    clearTimeout(this.timeout);
  };

  setRef = (ref: HTMLInputElement | null) => {
    if (this.state.ref !== ref) {
      this.setState({
        ref,
      });
    }
  };

  close = () => {
    const { ref } = this.state;
    ref && ref.blur();
    this.props.onClose();
  };

  render() {
    const {
      isExpanded,
      theme,
      children,
      sharedClient,
      selectedTabIndex,
      products,
      primaryProduct = '',
    } = this.props;

    if (products.length === 0) {
      return <ProductSearchInputSkeleton />;
    }

    return (
      <ProductProvider
        products={products}
        activeIndex={selectedTabIndex}
        primaryProduct={primaryProduct}
      >
        <MessagesIntlProvider>
          <BuildVersionAnalyticContext>
            <ThemeProvider partialSearchCSS={theme}>
              <SearchAnchor
                onBlur={this.onBlurHandler}
                onFocus={this.onFocusHandler}
                onKeyDown={this.handleKeyDown}
                isExpanded={isExpanded}
              >
                <KeyboardHighlightProvider listenerNode={this.state.ref}>
                  <ABTestProvider
                    searchClient={sharedClient}
                    isMultiProduct={products.length > 1}
                  >
                    <SearchSessionProvider
                      sessionKey={`ssid_provider_${isExpanded}`}
                    >
                      {isExpanded ? <ExperimentExposureHandler /> : null}
                      {children ? children(this.setRef) : null}
                    </SearchSessionProvider>
                  </ABTestProvider>
                </KeyboardHighlightProvider>
              </SearchAnchor>
            </ThemeProvider>
          </BuildVersionAnalyticContext>
        </MessagesIntlProvider>
      </ProductProvider>
    );
  }
}

export const ProductSearchDialog = (props: Props) => {
  const [permissibleProducts, setPermissibleProducts] = useState<Products[]>(
    props.products.length === 1 || !props.doProductPermissionsCheck
      ? props.products
      : [],
  );

  useEffect(() => {
    let cancelled = false;
    if (props.doProductPermissionsCheck && props.products.length !== 1) {
      fetchPermissions(props.sharedClient, props.products).then((products) => {
        !cancelled && setPermissibleProducts(products);
      });
    }

    return () => {
      cancelled = true;
    };
  }, [props.products, props.sharedClient, props.doProductPermissionsCheck]);

  return <ProductSearchDialogBase {...props} products={permissibleProducts} />;
};
