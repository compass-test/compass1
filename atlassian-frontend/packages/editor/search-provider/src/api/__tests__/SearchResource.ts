import fetchMock from 'fetch-mock/es5/client';
import { Scope } from '@atlassian/product-search-dialog/search-client';

import SearchResource from '../SearchResource';
import confluencePageBlogMock from '../__fixtures__/confluence-page-blog.json';
import emptyConfluencePageBlogMock from '../__fixtures__/empty-confluence-page-blog.json';
import jiraIssueMock from '../__fixtures__/jira-issue.json';
import emptyJiraIssueMock from '../__fixtures__/empty-jira-issue.json';

const searchUrl = '/gateway/api/xpsearch-aggregator/quicksearch/v1';
const cloudId = 'DUMMY-158c8204-ff3b-47c2-adbb-a0906ccc722b';

describe('SearchResource', () => {
  afterEach(fetchMock.restore);

  it('should throw error if unsupported Scope is passed in', () => {
    expect(() => {
      new SearchResource(cloudId, Scope.ConfluencePageBlogAttachment);
    }).toThrowErrorMatchingInlineSnapshot(
      `"Unsupported scope - confluence.page,blogpost,attachment"`,
    );
  });

  it('should return "" as default Ari if the information cannot correctly compose an Ari', async () => {
    const provider = new SearchResource(
      'broken-cloud-id',
      Scope.ConfluencePageBlog,
    );
    fetchMock.mock({
      matcher: searchUrl,
      response: confluencePageBlogMock,
      name: 'quick-search',
    });

    const result = await provider.quickSearch('abc', 5);

    expect(result).toMatchSnapshot();
  });

  it('should search for Scope.ConfluencePageBlog', async () => {
    const provider = new SearchResource(cloudId, Scope.ConfluencePageBlog);
    fetchMock.mock({
      matcher: searchUrl,
      response: confluencePageBlogMock,
      name: 'quick-search',
    });

    const result = await provider.quickSearch('abc', 5);

    expect(result).toMatchSnapshot();
  });

  it('should handle empty result for Scope.ConfluencePageBlog', async () => {
    const provider = new SearchResource(cloudId, Scope.ConfluencePageBlog);
    fetchMock.mock({
      matcher: searchUrl,
      response: emptyConfluencePageBlogMock,
      name: 'quick-search',
    });

    const result = await provider.quickSearch('abc', 5);

    expect(result).toMatchInlineSnapshot(`Array []`);
  });

  it('should search for Scope.JiraIssue', async () => {
    const provider = new SearchResource(cloudId, Scope.JiraIssue);
    fetchMock.mock({
      matcher: searchUrl,
      response: jiraIssueMock,
      name: 'quick-search',
    });

    const result = await provider.quickSearch('abc', 5);

    expect(result).toMatchSnapshot();
  });

  it('should handle empty result for Scope.JiraIssue', async () => {
    const provider = new SearchResource(cloudId, Scope.JiraIssue);
    fetchMock.mock({
      matcher: searchUrl,
      response: emptyJiraIssueMock,
      name: 'quick-search',
    });

    const result = await provider.quickSearch('abc', 5);

    expect(result).toMatchInlineSnapshot(`Array []`);
  });

  it('should throw error in quick search if Scope is not supported', async () => {
    const provider = new SearchResource(cloudId, Scope.JiraIssue);
    fetchMock.mock({
      matcher: searchUrl,
      response: jiraIssueMock,
      name: 'quick-search',
    });

    const bindedQuickSearch = provider.quickSearch.bind({
      ...provider,
      scope: 'nothing',
    });

    expect(bindedQuickSearch('abc', 5)).rejects.toMatchInlineSnapshot(
      `[Error: Unsupported scope - nothing]`,
    );
  });
});
