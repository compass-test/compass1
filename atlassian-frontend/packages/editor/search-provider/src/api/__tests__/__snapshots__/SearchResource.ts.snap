// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResource should return "" as default Ari if the information cannot correctly compose an Ari 1`] = `
Array [
  Object {
    "container": "Editor Media",
    "contentType": "confluence.page",
    "objectId": "",
    "title": "Editor Media Planning - 2020-07-07",
    "updatedTimestamp": "2020-07-07T03:24:39+00:00",
    "url": "https://product-fabric.atlassian.net/wiki/spaces/EM/pages/1538426521/Editor+Media+Planning+-+2020-07-07",
  },
  Object {
    "container": "Media Platform Team",
    "contentType": "confluence.page",
    "objectId": "",
    "title": "Weekly Media Triad Meeting",
    "updatedTimestamp": "2020-07-05T23:30:42+00:00",
    "url": "https://product-fabric.atlassian.net/wiki/spaces/FIL/pages/873989347/Weekly+Media+Triad+Meeting",
  },
  Object {
    "container": "Editor Media",
    "contentType": "confluence.page",
    "objectId": "",
    "title": "Media test Media single",
    "updatedTimestamp": "2020-05-05T06:06:28+00:00",
    "url": "https://product-fabric.atlassian.net/wiki/spaces/EM/pages/1356074858/Media+test+Media+single",
  },
  Object {
    "container": "Editor Media",
    "contentType": "confluence.page",
    "objectId": "",
    "title": "Editor Media OKRs",
    "updatedTimestamp": "2020-06-09T02:21:27+00:00",
    "url": "https://product-fabric.atlassian.net/wiki/spaces/EM/pages/1385631549/Editor+Media+OKRs",
  },
  Object {
    "container": "Editor",
    "contentType": "default",
    "objectId": "",
    "title": "Media linking",
    "updatedTimestamp": "2020-04-15T03:00:05+00:00",
    "url": "https://product-fabric.atlassian.net/wiki/spaces/E/pages/998082697/Media+linking",
  },
  Object {
    "container": "Editor",
    "contentType": "confluence.blogpost",
    "objectId": "",
    "title": "Media linking",
    "updatedTimestamp": "2020-04-15T03:00:05+00:00",
    "url": "https://product-fabric.atlassian.net/wiki/spaces/E/pages/998082697/Media+linking",
  },
]
`;

exports[`SearchResource should search for Scope.ConfluencePageBlog 1`] = `
Array [
  Object {
    "container": "Editor Media",
    "contentType": "confluence.page",
    "objectId": "ari:cloud:confluence:DUMMY-158c8204-ff3b-47c2-adbb-a0906ccc722b:content/1538426521",
    "title": "Editor Media Planning - 2020-07-07",
    "updatedTimestamp": "2020-07-07T03:24:39+00:00",
    "url": "https://product-fabric.atlassian.net/wiki/spaces/EM/pages/1538426521/Editor+Media+Planning+-+2020-07-07",
  },
  Object {
    "container": "Media Platform Team",
    "contentType": "confluence.page",
    "objectId": "ari:cloud:confluence:DUMMY-158c8204-ff3b-47c2-adbb-a0906ccc722b:content/873989347",
    "title": "Weekly Media Triad Meeting",
    "updatedTimestamp": "2020-07-05T23:30:42+00:00",
    "url": "https://product-fabric.atlassian.net/wiki/spaces/FIL/pages/873989347/Weekly+Media+Triad+Meeting",
  },
  Object {
    "container": "Editor Media",
    "contentType": "confluence.page",
    "objectId": "ari:cloud:confluence:DUMMY-158c8204-ff3b-47c2-adbb-a0906ccc722b:content/1356074858",
    "title": "Media test Media single",
    "updatedTimestamp": "2020-05-05T06:06:28+00:00",
    "url": "https://product-fabric.atlassian.net/wiki/spaces/EM/pages/1356074858/Media+test+Media+single",
  },
  Object {
    "container": "Editor Media",
    "contentType": "confluence.page",
    "objectId": "ari:cloud:confluence:DUMMY-158c8204-ff3b-47c2-adbb-a0906ccc722b:content/1385631549",
    "title": "Editor Media OKRs",
    "updatedTimestamp": "2020-06-09T02:21:27+00:00",
    "url": "https://product-fabric.atlassian.net/wiki/spaces/EM/pages/1385631549/Editor+Media+OKRs",
  },
  Object {
    "container": "Editor",
    "contentType": "default",
    "objectId": "ari:cloud:confluence:DUMMY-158c8204-ff3b-47c2-adbb-a0906ccc722b:content/998082697",
    "title": "Media linking",
    "updatedTimestamp": "2020-04-15T03:00:05+00:00",
    "url": "https://product-fabric.atlassian.net/wiki/spaces/E/pages/998082697/Media+linking",
  },
  Object {
    "container": "Editor",
    "contentType": "confluence.blogpost",
    "objectId": "ari:cloud:confluence:DUMMY-158c8204-ff3b-47c2-adbb-a0906ccc722b:content/998082697",
    "title": "Media linking",
    "updatedTimestamp": "2020-04-15T03:00:05+00:00",
    "url": "https://product-fabric.atlassian.net/wiki/spaces/E/pages/998082697/Media+linking",
  },
]
`;

exports[`SearchResource should search for Scope.JiraIssue 1`] = `
Array [
  Object {
    "container": "Editor Media",
    "contentType": "jira.issue.task",
    "objectId": "ari:cloud:jira:DUMMY-158c8204-ff3b-47c2-adbb-a0906ccc722b:issue/128578",
    "title": "FE error causing publishing errors - External Media Node must have a 'url' attribute",
    "updatedTimestamp": "2020-07-08T14:59:56.112+1000",
    "url": "https://product-fabric.atlassian.net/browse/EDM-627",
  },
  Object {
    "container": "Confluence Editor - Reactor Core",
    "contentType": "jira.issue.bug",
    "objectId": "ari:cloud:jira:DUMMY-158c8204-ff3b-47c2-adbb-a0906ccc722b:issue/162716",
    "title": "[BE] [SLA] [Edit Page] External media nodes are missing 'url' attribute",
    "updatedTimestamp": "2020-07-08T00:43:47.608+1000",
    "url": "https://product-fabric.atlassian.net/browse/CERN-374",
  },
  Object {
    "container": "",
    "contentType": "jira.issue.story",
    "objectId": "ari:cloud:jira:DUMMY-158c8204-ff3b-47c2-adbb-a0906ccc722b:issue/167819",
    "title": "Media file fails to insert",
    "updatedTimestamp": "2020-07-08T09:02:18.334+1000",
    "url": "https://product-fabric.atlassian.net/browse/EDM-833",
  },
  Object {
    "container": "Editor Media",
    "contentType": "jira.issue.task",
    "objectId": "ari:cloud:jira:DUMMY-158c8204-ff3b-47c2-adbb-a0906ccc722b:issue/169212",
    "title": "Create Editor Media amplitude dashboard",
    "updatedTimestamp": "2020-07-08T09:02:18.548+1000",
    "url": "https://product-fabric.atlassian.net/browse/EDM-848",
  },
  Object {
    "container": "Editor Media",
    "contentType": "jira.issue",
    "objectId": "ari:cloud:jira:DUMMY-158c8204-ff3b-47c2-adbb-a0906ccc722b:issue/170464",
    "title": "We've had a bunch of performance issues in Smart Links - all fixed, but we should monitor/instrument them (4 votes)",
    "updatedTimestamp": "2020-06-19T16:00:12.589+1000",
    "url": "https://product-fabric.atlassian.net/browse/EDM-888",
  },
]
`;
